core/src/main/java/org/elasticsearch/ingest/processor/ConvertProcessor.java

```diff
@@ -138,7 +141,13 @@ public String getType() {
         @Override
         public ConvertProcessor doCreate(String processorTag, Map<String, Object> config) throws Exception {
             String field = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "field");
-            Type convertType = Type.fromString(ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "type"));
+            String typeProperty = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "type");
+            Type convertType;
+            try {
+                convertType = Type.fromString(typeProperty);
```

Comment:
Since we use `Type.fromString` only here maybe should change `Type.fromString()` so that it throws the right exception? We would have to change the signature to accept a tag.